plugins/memblaze/memblaze-nvme.c

```diff
@@ -1197,3 +1197,954 @@ static int mb_set_lat_stats(int argc, char **argv,
 	dev_close(dev);
 	return err;
 }
+
+// Global definitions
+
+static inline int K2C(int k) {  // KELVINS_2_CELSIUS
+    return (k - 273);
+};
+
+// Global ID definitions
+
+enum {
+    // feature ids
+    FID_LATENCY_FEATURE = 0xd0,
+
+    // log ids
+    LID_SMART_LOG_ADD          = 0xca,
+    LID_LATENCY_STATISTICS     = 0xd0,
+    LID_HIGH_LATENCY_LOG       = 0xd1,
+    LID_PERFORMANCE_STATISTICS = 0xd2,
+};
+
+// smart-log-add
+
+struct smart_log_add_item {
+    uint32_t index;
+    char    *attr;
+};
+
+#pragma pack(push, 1)
+struct wear_level {
+    __le16 min;
+    __le16 max;
+    __le16 avg;
+};
+
+struct smart_log_add_item_12 {
+    uint8_t id;
+    uint8_t rsvd[2];
+    uint8_t norm;
+    uint8_t rsvd1;
+    union {
+        struct wear_level wear_level;  // 0xad
+        struct temp_since_born {       // 0xe7
+            __le16 max;
+            __le16 min;
+            __le16 curr;
+        } temp_since_born;
+        struct power_consumption {  // 0xe8
+            __le16 max;
+            __le16 min;
+            __le16 curr;
+        } power_consumption;
+        struct temp_since_power_on {  // 0xaf
+            __le16 max;
+            __le16 min;
+            __le16 curr;
+        } temp_since_power_on;
+        uint8_t raw[6];
+    };
+    uint8_t rsvd2;
+};
+
+struct smart_log_add_item_10 {
+    uint8_t id;
+    uint8_t norm;
+    union {
+        struct wear_level wear_level;  // 0xad
+        uint8_t           raw[6];
+    };
+    uint8_t rsvd[2];
+};
+#pragma pack(pop)
```

Comment:
I suggest to use `__packed` attribute here. Those open coded pragmas are usually easy to miss.

Reply:
Got, I will create a new thread for this PR.